### DISCUSSION TITLE
Can the server be used with another client?

### DISCUSSION THREAD

**Discussion by kolserdav** [socketio/socket.io#4388] [Q&A] [answered]
I can't connect to socket.io server via browser native class WebSocket.

**To Reproduce**

Create basic socket.io Hello world and simple new WebSocket() connection in browser

Socket.IO server version: `4.5.1`

*Server*

```js
import express from 'express';
import http from 'http';
import { Server } from 'socket.io';

const app = express();

app.get('/', (req, res) => {
  res.send('<h1>Hello world</h1>');
});

const server = http.createServer(app);

const io = new Server(server);

io.on('connection', (socket) => {
  console.log('a user connected');
});

server.listen(parseInt(SERVER_PORT, 10), () => {
  console.log('listening on port', SERVER_PORT);
});
```

Socket.IO client version: `none`

*Client*

```js
const connection = new WebSocket(
      `${window.location.protocol === 'https' ? 'wss' : 'ws'}://${process.env.REACT_APP_SERVER}:${
        process.env.REACT_APP_PORT
      }`,
      'json'
    );
```

**Expected behavior**

**Platform:**
 - Device: [Desktop]
 - OS: [Arch Linux]

**Additional context**
- node.js [16.13.0]

**Comment by darrachequesne** [comment, marked as answer]
No, you won't be able to connect via plain WebSocket.

Reference: https://socket.io/docs/v4/#what-socketio-is-not

**Reply by kolserdav** [reply]
Thank You🙏
